src/atlas/meshgenerator/detail/CubedSphereDualMeshGenerator.cc

```diff
@@ -63,9 +63,7 @@ CubedSphereDualMeshGenerator::CubedSphereDualMeshGenerator(const eckit::Parametr
 
     // Get partitioner.
     std::string partitioner;
-    try { p.get("partitioner.type", partitioner); } catch( std::exception& ) {}
-    p.get("partitioner", partitioner);
-    if( partitioner.size() ) {
+    if (p.get("partitioner", partitioner) && partitioner.size()) {
         options.set("partitioner", partitioner);
```

Comment:
I believe I have another proposal that should work instead:
```
    // Get partitioner.
    if (p.has("partitioner.type")) {
        std::string partitioner;
        p.get("partitioner.type", partitioner);
        options.set("partitioner", partitioner);
    }
    else if (p.has("partitioner")) {
        std::string partitioner;
        p.get("partitioner", partitioner);
        options.set("partitioner", partitioner);
    }
 ```
 Could you test this change instead and get back to me if that also solves the issue?

Reply:
That looks far more civilised than juggling pointers...

Reply:
Thank you, @wdeconinck. I've tested this, extensively. Unfortunately, it doesn't solve the problem at all. It seems that the "has" function performs something of an unsympathetic search that continues to generate the underlying exception in eckit we're trying to avoid. Removal of the try-catch, as in your suggestion therefore results in the exception terminating the running program/test.

I've found a slight modification that works for us:

`    try {
        if (p.has("partitioner")) {
            std::string partitioner;
            p.get("partitioner", partitioner);
            options.set("partitioner", partitioner);
        }
        else if (p.has("partitioner.type")) {
            std::string partitioner;
            p.get("partitioner.type", partitioner);
            options.set("partitioner", partitioner);
        }
    } catch( std::exception& ) {}`
    
This wraps your suggestion, @wdeconinck, in a try-catch and swaps the if clauses so that our current configuration is successfully found and the exception is avoided. However, this will obviously generate the exception message for users who have moved to this new "partitioner.type" use-case.

Is it possible you could show us an example of this use-case being used as intended? As described above, simply swapping our call to the `atlas::MeshGenerator` to use the handle "partitioner.type" instead of "partitioner" (as we do now) seems to introduce a  different problem.

Reply:
Following not yet merged [commit](https://github.com/ecmwf/eckit/pull/101/commits/5eddd335b6aaf1b6f326b97a59f8e40b4a905087)  in eckit would make my latest suggestion work in the future.

I cannot give an existing use case for this @phlndrwd. It could have been as part of some testing in another project that I don't remember. It also does not seem to be added to the other MeshGenerators strangely. We can for now accept @matthewrmshin and reintroduce (if needed) support for "partitioner.type", which would then also require changes in other mesh generators.

Reply:
Thanks @wdeconinck! I certainly prefer a long-lasting solution and I'm happy to hear that you guys are already on it :)